### ISSUE TITLE
Data import fails when system locale uses dot as time separator

### ISSUE THREAD

**Issue by plindman** [furore-fhir/spark#103]
When running Maintenance/Initialize, there are multiple errors when system locale time separator is dot ("."). Example error message: `ERROR Importing Account example... : The string '2016-10-14T14.37.44+03:00' is not a valid AllXsd value.`

Accepts time string like: `"2015-10-24T07:41:03+11:00"`
Fails time string line like: `"2016-10-14T16.13.09+03:00"`

In Spark.Engine\Search\ElementIndexer.cs row 170 the call to LowerBound: 
`bounds.Add(new IndexValue("start", new DateTimeValue(element.LowerBound())));`

Error thrown from Spark.Engine\Extensions\FhirDateTimeExtensions.cs row 51: 

```
public static DateTimeOffset LowerBound(this FhirDateTime fdt)
{
    return fdt.ToDateTimeOffset();
}
```

**Comment by plindman** [comment]
Seems that  [FhirDateTime](https://github.com/ewoutkramer/fhir-net-api/blob/master/src/Hl7.Fhir.Core/Model/FhirDateTime.cs) is the culprit because it uses DateTimeOffset without any CultureInfo to create the string Value (which creates a locale-specific string) but then uses XmlConvert to convert back to DateTimeOffset which requires an CultureInvariant string (locale formatted string causes FormatException).

```
DateTimeOffset.Now.ToString("yyyy-MM-dd'T'HH:mm:ss.fffK", System.Globalization.CultureInfo.InvariantCulture));
DateTimeOffset.Parse("2016-10-10T19:00:02+01:00", System.Globalization.CultureInfo.InvariantCulture));
```

As an ugly quick-fix `fdt.Value = System.Text.RegularExpressions.Regex.Replace(fdt.Value, "[.](?=\\d{2}?)(?!\\d{3}?)", ":");` added to LowerBound fixes the problem temporarily (the regex changes minutes and seconds dots but leaves milliseconds dot in place).

**Comment by plindman** [comment]
Discussed in fhir-net-api [issues/256](https://github.com/ewoutkramer/fhir-net-api/issues/256) and fixed in [365fc99a0c48db68e7f78fa13e2f3243d8791091](https://github.com/ewoutkramer/fhir-net-api/commit/365fc99a0c48db68e7f78fa13e2f3243d8791091).
